python/lsst/ts/Dome/dome_csc.py

```diff
@@ -178,8 +178,6 @@ async def write_then_read_reply(self, cmd):
             A dict of the form {"reply": {"param1": value1, "param2": value2}} where "reply" can for
             instance be "OK" or "ERROR".
          """
-        # TODO Create a function that takes the command and its parameters and returns a yaml
-        #  representation of it so all do_XXX functions below can be simplified. DM-24776
```

Comment:
What I had in mind is to change this method to take the command and parameters, so the user doesn't have to know the details of the form sent to the low-level controller:
```
def write_then_read_reply(self, cmd, **params):
    st = yaml.safe_dump({cmd: **params}, default_flow_style=None)
    self.writer.write(st.encode() + b"\r\n")
```
Here are two examples of usage:
```
await self.write_then_read_reply("crawlEl", dirMotion=data.dirMotion, elRate=math.radians(data.elRate))
await self.write_then_read_reply("stopEl")
```

Reply:
Thanks for the explanation. We will replace YAML with JSON soon so I will take care of this then. See

https://jira.lsstcorp.org/browse/DM-25130

Reply:
Done

Reply:
Thanks. I'm glad you found it useful.